src/main/java/org/elasticsearch/search/fetch/termvectors/TermVectorsBuilder.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.fetch.termvectors;
+
+import org.elasticsearch.action.termvectors.TermVectorsRequest;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ */
+public class TermVectorsBuilder implements ToXContent {
+
+    private Boolean fetchOnly;
+    private String[] selectedFields;
+    private Boolean realtime;
+    private Map<String, String> perFieldAnalyzer;
+    private Boolean positions;
+    private Boolean offsets;
+    private Boolean payloads;
+    private Boolean fieldStatistics;
+    private Boolean termStatistics;
+    private Boolean dfs;
+    private TermVectorsRequest.FilterSettings filterSettings;
+
+    /**
+     * Specifies whether to return the stored term vectors, disregarding any previous parameters.
+     */
+    public TermVectorsBuilder setFetchOnly(Boolean fetchOnly) {
+        this.fetchOnly = fetchOnly;
+        return this;
+    }
```

Comment:
The name of this parameter and how it hides every other parameter feels awkward to me.